lint/rule_test.go

```diff
@@ -0,0 +1,396 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lint
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/builder"
+)
+
+func TestFileRule(t *testing.T) {
+	// Create a file descriptor with nothing in it.
+	fd, err := builder.NewFile("test.proto").Build()
+	if err != nil {
+		t.Fatalf("Could not build file descriptor.")
+	}
+
+	// Iterate over the tests and run them.
+	for _, test := range makeLintRuleTests(fd) {
+		t.Run(test.testName, func(t *testing.T) {
+			uri := fmt.Sprintf("https://aip.dev/%s", t.Name())
+			rule := &FileRule{
+				Name: NewRuleName("test", test.testName),
+				URI:  uri,
+				LintFile: func(fd *desc.FileDescriptor) []Problem {
+					return test.problems
+				},
+			}
+
+			// Run the rule and assert that we got what we expect.
+			test.runRule(rule, fd, t)
+		})
+	}
+}
+
+func TestMessageRule(t *testing.T) {
+	// Create a file descriptor with two messages in it.
+	fd, err := builder.NewFile("test.proto").AddMessage(
+		builder.NewMessage("Book"),
+	).AddMessage(
+		builder.NewMessage("Author"),
+	).Build()
+	if err != nil {
+		t.Fatalf("Failed to build file descriptor.")
```

Comment:
Print the `err` as well.

Reply:
Done, but this really does not matter as there is no user input or other variance, and this will never actually error.